src/backend/utils/error/elog.c

```diff
@@ -5353,59 +5335,9 @@ debug_backtrace(void)
  */
 uint32 gp_backtrace(void **stackAddresses, uint32 maxStackDepth)
 {
-#if defined(__i386) || defined(__x86_64__)
-
-	/*
-	 * Stack base pointer has not been initialized by PostmasterMain,
-	 * or PostgresMain/AuxiliaryProcessMain is called directly by main
-	 * rather than forked by PostmasterMain (such as when initdb).
-	 *
-	 * In this case, just return depth as 0 to indicate that we have not
-	 * stored any frame addresses.
-	 */
-	if (stack_base_ptr == NULL)
-		return 0;
-
-	/* get base pointer of current frame */
-	uint64 framePtrValue = 0;
-	GET_FRAME_POINTER(framePtrValue);
-
-	uint32 depth = 0;
-	void **pFramePtr = (void**) GET_PTR_FROM_VALUE(framePtrValue);
-
-	/* check if the frame pointer is valid */
-	if (pFramePtr != NULL && (void *) &depth < (void *) pFramePtr)
-	{
-		/* consider the first maxStackDepth frames only, below the stack base pointer */
-		for (depth = 0; depth < maxStackDepth; depth++)
-		{
-			/* check if next frame is within stack */
-			if (pFramePtr == NULL ||
-				(void *) pFramePtr > *pFramePtr ||
-				(void *) stack_base_ptr < *pFramePtr)
-			{
-				break;
-			}
-
-			/* get return address (one above the frame pointer) */
-			const uintptr_t *returnAddr = (uintptr_t *)(pFramePtr + 1);
-
-			/* store return address */
-			stackAddresses[depth] = (void *) *returnAddr;
-
-			/* move to next frame */
-			pFramePtr = (void**)*pFramePtr;
-		}
-	}
-	else
-	{
-		depth  = backtrace(stackAddresses, maxStackDepth);
-	}
-
-	Assert(depth > 0);
-
-	return depth;
-
+#ifdef WIN32
+	/* backtrace is not available for this platform */
+	return 0;
```

Comment:
why do use WIN32 branch? IMO gpdb in general doesn't support windows

Reply:
I agree with you, but mostly it was done because [gp_stacktrace()](https://github.com/arenadata/gpdb/blob/c175f22b9f382eaeb8378399ac6714fc4267957b/src/backend/utils/error/elog.c#L5355) does the same.